src/wxdoughnutchart.cpp

```diff
@@ -28,12 +28,24 @@ wxDoughnutChartData::wxDoughnutChartData()
 
 void wxDoughnutChartData::AppendSlice(const wxChartSliceData &slice)
 {
-    m_slices.push_back(slice);
```

Comment:
This is not ideal. Let's make the m_value of the base class protected so it can be directly updated here. And then call OnUpdate here too. I know it breaks encapsulation on the base class but the base class is quite small so it's OK.